src/top_model/main.cpp

```diff
@@ -209,5 +214,12 @@ std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> TOP = std::make_share
     r.run_until(NDTime("04:00:00:000"));
     auto elapsed = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
     cout << "Simulation took:" << elapsed << "sec" << endl;
+	
+	FILE *input;
+	FILE *output;
+	input = fopen("../data/abp_output.txt","r");
+	output = fopen("../data/abp_processed_output.csv","w");
+	filter(input,output);
```

Comment:
We were asked to just pass the file paths in function as parameters. So as per that instruction, just pass the file paths, and open the file in the relevant function. Also, kindly write comments so that the reader can understand it.